client/orderbook/orderbook.go

```diff
@@ -326,17 +364,24 @@ func (ob *OrderBook) updateRemaining(note *msgjson.UpdateRemainingNote, cached b
 	var oid order.OrderID
 	copy(oid[:], note.OrderID)
 
-	ord := ob.sells.UpdateRemaining(oid, note.Remaining)
-	if ord != nil {
-		return nil
+	ob.ordersMtx.Lock()
+	ord, found := ob.orders[oid]
+	if found {
+		newOrder := ord.copy()
+		newOrder.Quantity = note.Remaining
+		ob.orders[oid] = newOrder
 	}
-
-	ord = ob.buys.UpdateRemaining(oid, note.Remaining)
-	if ord != nil {
-		return nil
+	ob.ordersMtx.Unlock()
```

Comment:
I'm a little unsure about the need to replace the order in the map with a fresh Order instance, but also the need to do anything with the `orders` map at all since the two booksides refer to the map entries (pointers).
Did you observe a bug or did this just seem like a needed change?

Reply:
Curious that the race detector hasn't hit it before.  Maybe because when that code is called in bookie.go we've always been locking `booky.mtx`?